src/item_factory.cpp

```diff
@@ -176,9 +176,11 @@ void Item_factory::finalize_pre( itype &obj )
         obj.integral_volume = obj.volume;
     }
     // for ammo and comestibles stack size defaults to count of initial charges
+    // Set max stack size to 200 to prevent integer overflow
```

Comment:
Should we throw a debug message every time something tries to set stack sizes and container volumes above the limit?

Reply:
The current code has a logic error. I fixed it and did some debugging/testing, and it does act weird when the stack size changes. The volume of the same stack of gold gets multiplied by 5. How should we set a max stack size and avoid causing this problem?

Reply:
I'd be fine with debugmsg + truncate the value to the acceptable range.
It's just having it fully silent is the problem.

AFAIK the stack size change needs to be accompanied by adjusting the accompanying stats to make it work out.
For example with gold it seems like it needs to have volume 1mL, weight 20g for a single unit of "gold", and the stack size should be whatever is most convinient, probably 200 since that's a rather small unit.  Count should be adjusted proportionally with the adjustment to grams in order to make it spawn in the same amounts.

Reply:
Changed the stack sizes for items above 200 to 200, and changed the volumes to match. Because the stack size only affects volume, I kept the weight and counts the same so that the density of each charge remained the same, and wouldn't affect crafting recipes.